File: server/mvc/routes/messaging.js
const { Router } = require("express");
const {
    getConversations,
    getMessages,
    getUnreadMessages,
    startConversation,
} = require("../controllers/messaging");
const {
    authorizeUser,
    getLimit,
    postLimit,
} = require("./middleware/middleware");

const router = Router();

router.post("/startConversation", postLimit, authorizeUser, startConversation);

router.get("/getConversations", getLimit, authorizeUser, getConversations);
router.get(
    "/getMessages/:conversationId",
    getLimit,
    authorizeUser,
    getMessages
);
router.get("/getUnreadMessages", getLimit, authorizeUser, getUnreadMessages);

module.exports = router;

File: server/routes/messaging.go
package routes

import "github.com/gorilla/mux"

func RegisterMessagingRoutes(router *mux.Router) {
	messagingRouter := router.PathPrefix("/messaging").Subrouter()

	messagingRouter.HandleFunc("/startConversation", StartConversation).Methods("POST")
	messagingRouter.HandleFunc("/deleteMessage", DeleteMessage).Methods("POST")

	messagingRouter.HandleFunc("/getConversations/{page}", GetConversations).Methods("GET")
	messagingRouter.HandleFunc("/getMessages/{conversationId}/{page}", GetMessages).Methods("GET")
	messagingRouter.HandleFunc("/getUnreadMessages", GetUnreadMessages).Methods("GET")
}
